spec.bs

```diff
@@ -252,19 +252,216 @@ dictionary AuctionAdConfig {
+    |component| and [=validate and convert auction ad config/isTopLevel=] set to false.
+  1. [=list/Append=] |componentAuction| to |auctionConfig|'s [=auction config/component auctions=].
+1. If |config|["{{AuctionAdConfig/sellerExperimentGroupId}}"] [=map/exists=]:
+  1. Set |auctionConfig|'s [=auction config/seller experiment group id=] to
+    |config|["{{AuctionAdConfig/sellerExperimentGroupId}}"].
+1. Return |auctionConfig|.
+
+</div>
+
+To <dfn>parse an origin</dfn> given a [=string=] |input|:
+1. Let |url| be the result of running the [=URL parser=] on |input|.
+1. If |url| is an error, then return failure.
+1. Return |url|'s [=url/origin=].
+
+To <dfn>start bidding and scoring</dfn> given an [=auction config=] |auctionConfig|:
+1. If |auctionConfig|'s [=auction config/component auctions=] are empty:
+  1. TODO: Create seller worklet.
+1. OtherWise:
+  1. [=list/For each=] |component| in |auctionConfig|'s [=auction config/component auctions=]:
+    1. [=Start bidding and scoring=] with |component|.
+1. Create bidder worklets [=in parallel=]. [=list/For each=] |buyer| in
+  [=auction config/interest group buyers=]:
+  1. Let |bidderWorklet| be a [=bidder worklet=] object for |buyer|. TODO: expand.
+  1. Let |biddingScript| be the result of [=fetching a bidding javascript=] with |bidderWorklet|.
+  1. If |biddingScript| is an error, [=iteration/continue=].
+  1. TODO: Run all [=generate a bid|GenerateBid()=] tasks that are ready.
+  1. TODO: Maybe rewrite this part.
+
+To <dfn>create a request</dfn> given a [=url=] |url|, and a [=string=] <dfn for="create a request">
+|accept|</dfn>:
+1. Let |request| be a new [=request=] with the following properties:
+    :   [=request/URL=]
+    ::  |url|
+    :   [=request/header list=]
+    ::  A new [=header list=] containing a [=header=] named "`Accept`" whose value is |accept|
+    :   [=request/redirect mode=]
+    :: "`error`"
+    :   [=request/referrer=]
+    :: "`no-referrer`"
+    :   [=request/client=]
+    ::  `null`
+    :   [=request/window=]
+    ::  "`no-window`" TODO: verify
+    :   [=request/service-workers mode=]
+    ::  "`none`"
+    :   [=request/mode=]
+    ::  "`cors`"
+    :   [=request/credentials mode=]
+    ::  "`omit`"
+    :   [=request/cache mode=]
+    ::  "`no-store`"
+1. Return |request|.
+
+To <dfn>fetch a bidding javascript</dfn> with given [=bidder worklet=] |bidderWorklet|:
+1. Let |request| be the result of [=creating a request=] with |bidderWorklet|'s
+    [=bidder worklet/bidding url=], and [=create a request/accept=] set to
+    "`application/javascript`".
+1. [=Fetch=] |request|.
+1. Let |response| be the result of [=fetch=] when it asynchronously completes.
+1. Let |body| be |response|’s [=response/body=].
+1. If |body| is null, return failure.
+1. Let |headers| be |response|'s [=response/header list=].
+1. If |headers| is empty or [=header list/get=] "X-Allow-FLEDGE" from |headers| does not return

```

Comment:
I think "get" on an empty header list will return null, so you don't need the first condition:
```suggestion
1. If [=header list/get=] "X-Allow-FLEDGE" from |headers| does not return
```